src/hb-aat.h

```diff
@@ -0,0 +1,765 @@
+  HB_AAT_LAYOUT_SELECTOR_STYLISTIC_ALT_THIRTEEN_OFF  = 27,
+  HB_AAT_LAYOUT_SELECTOR_STYLISTIC_ALT_FOURTEEN_ON   = 28,
+  HB_AAT_LAYOUT_SELECTOR_STYLISTIC_ALT_FOURTEEN_OFF  = 29,
+  HB_AAT_LAYOUT_SELECTOR_STYLISTIC_ALT_FIFTEEN_ON    = 30,
+  HB_AAT_LAYOUT_SELECTOR_STYLISTIC_ALT_FIFTEEN_OFF   = 31,
+  HB_AAT_LAYOUT_SELECTOR_STYLISTIC_ALT_SIXTEEN_ON    = 32,
+  HB_AAT_LAYOUT_SELECTOR_STYLISTIC_ALT_SIXTEEN_OFF   = 33,
+  HB_AAT_LAYOUT_SELECTOR_STYLISTIC_ALT_SEVENTEEN_ON  = 34,
+  HB_AAT_LAYOUT_SELECTOR_STYLISTIC_ALT_SEVENTEEN_OFF = 35,
+  HB_AAT_LAYOUT_SELECTOR_STYLISTIC_ALT_EIGHTEEN_ON   = 36,
+  HB_AAT_LAYOUT_SELECTOR_STYLISTIC_ALT_EIGHTEEN_OFF  = 37,
+  HB_AAT_LAYOUT_SELECTOR_STYLISTIC_ALT_NINETEEN_ON   = 38,
+  HB_AAT_LAYOUT_SELECTOR_STYLISTIC_ALT_NINETEEN_OFF  = 39,
+  HB_AAT_LAYOUT_SELECTOR_STYLISTIC_ALT_TWENTY_ON     = 40,
+  HB_AAT_LAYOUT_SELECTOR_STYLISTIC_ALT_TWENTY_OFF    = 41
+};
+
+
+/*
+ * Selectors for HB_AAT_LAYOUT_FEATURE_TYPE_CONTEXTUAL_ALTERNATIVES
+ *
+ * Since: REPLACEME
+ */
+enum
+{
+  HB_AAT_LAYOUT_SELECTOR_CONTEXTUAL_ALTERNATES_ON        = 0,
+  HB_AAT_LAYOUT_SELECTOR_CONTEXTUAL_ALTERNATES_OFF       = 1,
+  HB_AAT_LAYOUT_SELECTOR_SWASH_ALTERNATES_ON             = 2,
+  HB_AAT_LAYOUT_SELECTOR_SWASH_ALTERNATES_OFF            = 3,
+  HB_AAT_LAYOUT_SELECTOR_CONTEXTUAL_SWASH_ALTERNATES_ON  = 4,
+  HB_AAT_LAYOUT_SELECTOR_CONTEXTUAL_SWASH_ALTERNATES_OFF = 5
+};
+
+
+/*
+ * Selectors for HB_AAT_LAYOUT_FEATURE_TYPE_LOWER_CASE
+ *
+ * Since: REPLACEME
+ */
+enum
+{
+  HB_AAT_LAYOUT_SELECTOR_DEFAULT_LOWER_CASE      = 0,
+  HB_AAT_LAYOUT_SELECTOR_LOWER_CASE_SMALL_CAPS   = 1,
+  HB_AAT_LAYOUT_SELECTOR_LOWER_CASE_PETITE_CAPS  = 2
+};
+
+
+/*
+ * Selectors for HB_AAT_LAYOUT_FEATURE_TYPE_UPPER_CASE
+ *
+ * Since: REPLACEME
+ */
+enum
+{
+  HB_AAT_LAYOUT_SELECTOR_DEFAULT_UPPER_CASE      = 0,
+  HB_AAT_LAYOUT_SELECTOR_UPPER_CASE_SMALL_CAPS   = 1,
+  HB_AAT_LAYOUT_SELECTOR_UPPER_CASE_PETITE_CAPS  = 2
+};
+
+
+/*
+ * Selectors for HB_AAT_LAYOUT_FEATURE_TYPE_CJK_ROMAN_SPACING_TYPE
+ *
+ * Since: REPLACEME
+ */
+enum
+{
+  HB_AAT_LAYOUT_SELECTOR_HALF_WIDTH_CJK_ROMAN    = 0,
+  HB_AAT_LAYOUT_SELECTOR_PROPORTIONAL_CJK_ROMAN  = 1,
+  HB_AAT_LAYOUT_SELECTOR_DEFAULT_CJK_ROMAN       = 2,
+  HB_AAT_LAYOUT_SELECTOR_FULL_WIDTH_CJK_ROMAN    = 3
+};
+
+
+/**
+ * hb_aat_layout_feature_record_t:
+ *
+ * Feature record and
+ *
+ * Since: REPLACEME
+ **/
+typedef struct hb_aat_layout_feature_record_t
+{
+  hb_aat_layout_feature_type_t feature;
+  hb_ot_name_id_t name_id;
+} hb_aat_layout_feature_record_t;
+
+HB_EXTERN unsigned int
+hb_aat_layout_get_features (hb_face_t                      *face,
+			    unsigned int                    start_offset,
+			    unsigned int                   *record_count, /* IN/OUT.  May be NULL. */
+			    hb_aat_layout_feature_record_t *record_buffer /* OUT.     May be NULL. */);
+
+/**
+ * hb_aat_feature_t:
+ *
+ * Feature value
+ *
+ * Since: REPLACEME
+ */
+typedef unsigned int hb_aat_layout_feature_setting_t;
```

Comment:
Maybe we put all the selectors in one enum and call it `hb_aat_layout_feature_setting_t`?  Guess I like that better.  The multiple anonymous enums currently don't add any benefit other than looking nice.

Reply:
Done